controllers/webhook.js

```diff
@@ -0,0 +1,260 @@
+  // Allow for multiple errors to be raised and reported back.
+  const errors = []
+
+  const ua = config.get('analytics.uaTrackingId')
+    ? require('universal-analytics')(config.get('analytics.uaTrackingId'))
+    : null
+
+  let gitService = null
+  let mergeReqNbr = null
+  if (service === 'github') {
+    /*
+     * In v1 of the endpoint, the service, username, repository, and branch parameters were
+     * ommitted. As such, if not provided in the webhook request URL, pull them from the webhook
+     * payload.
+     */
+    if (username === null || typeof username === 'undefined') {
+      username = data.repository.owner.login
+    }
+    if (repository === null || typeof repository === 'undefined') {
+      repository = data.repository.name
+    }
+    if (branch === null || typeof branch === 'undefined') {
+      branch = data.pull_request.base.ref
+    }
+
+    gitService = await gitFactory.create('github', {
+      version: version,
+      username: username,
+      repository: repository,
+      branch: branch
+    })
+    mergeReqNbr = data.number
+  } else if (service === 'gitlab') {
+    gitService = await gitFactory.create('gitlab', {
+      version: version,
+      username: username,
+      repository: repository,
+      branch: branch
+    })
+    mergeReqNbr = data.object_attributes.iid
+  } else {
+    errors.push('Unable to determine service.')
+    return Promise.reject(errors)
+  }
+
+  if (mergeReqNbr === null || typeof mergeReqNbr === 'undefined') {
+    errors.push('No pull/merge request number found.')
+    return Promise.reject(errors)
+  }
+
+  let review = await gitService.getReview(mergeReqNbr).catch((error) => {
+    const msg = `Failed to retrieve merge request ${mergeReqNbr} - ${error}`
+    console.error(msg)
+    errors.push(msg)
+    return Promise.reject(errors)
+  })
+
+  /*
+   * We might receive "real" (non-bot) pull requests for files other than Staticman-processed
+   * comments. Ignore these by filtering on Staticman-created merge request branches.
+   */
+  if (review.sourceBranch.indexOf('staticman_') > -1) {
+    /*
+     * Note that the review state is derived from the call to the git service, NOT the webhook
+     * payload. And merged requests are represented with different states in each (at least in
+     * GitHub). Tricky.
+     *
+     * We'll regularly receive webhook calls whenever a pull/merge request is opened, not just
+     * merged/closed.
+     */
+    if (review.state === 'merged' || review.state === 'closed') {
+      await _createNotifyMailingList(review, ua).catch((error) => {
+        errors.push(error.message)
+      })
+
+      /*
+       * Deleting the merge request branch is only necessary for GitHub, as GitLab automatically
+       * deletes it upon merging.
+       */
+      if (service === 'github') {
+        await _deleteMergeRequestBranch(gitService, review, ua).catch((error) => {
+          errors.push(error)
+        })
+      }
```

Comment:
```suggestion
      if (_requiresManualDelete(service)) {
        await _deleteMergeRequestBranch(gitService, review, ua).catch((error) => {
          errors.push(error)
        })
      }
```

Comment could be a helper function like `_requiresManualDelete()`

Reply:
Changed as requested.